test/unit/sqs/index.test.ts

```diff
@@ -22,140 +23,296 @@ describe("SqsQueue tests", () => {
 			queueRegion: TEST_QUEUE_REGION,
-			longPollingIntervalSec: 0
+			longPollingIntervalSec: 0,
+			timeoutSec: timeout,
+			maxAttempts: 3
 		},
-		mockRequestHandler);
+		mockRequestHandler,
+		mockErrorHandler);
 	};
 
 	let queue: SqsQueue<TestMessage>;
 
-	beforeEach(() => {
-		queue = createSqsQueue();
-		queue.start();
-	});
+	describe("Normal execution tests", () => {
 
-	afterEach(async () => {
-		queue.stop();
-		await queue.waitUntilListenerStopped();
-	});
+		beforeEach(() => {
+			queue = createSqsQueue(10);
+			queue.start();
+		});
 
-	test("Message gets received", (done: DoneCallback) => {
+		afterEach(async () => {
+			queue.stop();
+			await queue.waitUntilListenerStopped();
+		});
+
+		test("Message gets received", (done: DoneCallback) => {
 
-		const testPayload = generatePayload();
+			const testPayload = generatePayload();
 
-		mockRequestHandler.handle.mockImplementation((context: Context<TestMessage>) => {
-			expect(context.payload).toStrictEqual(testPayload);
-			done();
+			mockRequestHandler.handle.mockImplementation((context: Context<TestMessage>) => {
+				expect(context.payload).toStrictEqual(testPayload);
+				done();
+			});
+			queue.sendMessage(testPayload);
 		});
-		queue.sendMessage(testPayload);
-	});
 
 
-	test("Queue is restartable", async (done: DoneCallback) => {
+		test("Queue is restartable", async (done: DoneCallback) => {
 
-		const testPayload = generatePayload();
+			const testPayload = generatePayload();
 
-		mockRequestHandler.handle.mockImplementation((context: Context<TestMessage>) => {
-			expect(context.payload).toStrictEqual(testPayload);
-			done();
+			mockRequestHandler.handle.mockImplementation((context: Context<TestMessage>) => {
+				expect(context.payload).toStrictEqual(testPayload);
+				done();
+			});
+
+			queue.stop();
+
+			//delaying to make sure all asynchronous invocations inside the queue will be finished and it will stop
+			await queue.waitUntilListenerStopped();
+
+			queue.start();
+
+			queue.sendMessage(testPayload);
 		});
 
-		queue.stop();
+		test("Message received with delay", (done: DoneCallback) => {
 
-		//delaying to make sure all asynchronous invocations inside the queue will be finished and it will stop
-		await queue.waitUntilListenerStopped();
+			const testPayload = generatePayload();
+			const receivedTime = {time: new Date().getTime()};
 
-		queue.start();
+			mockRequestHandler.handle.mockImplementation((context: Context<TestMessage>) => {
 
-		queue.sendMessage(testPayload);
-	});
+				if(context.payload.msg !== testPayload.msg) {
+					//discard message if it is not our test message
+					return;
+				}
+
+				try {
+					context.log.info("hi");
+					const currentTime = new Date().getTime();
+					expect(currentTime - receivedTime.time).toBeGreaterThanOrEqual(1000);
+					done();
+				} catch (err) {
+					done(err);
+				}
+			});
+			queue.sendMessage(testPayload, 1);
+		});
+
+		test("Message gets executed exactly once", (done: DoneCallback) => {
```

Comment:
Jest/JS standard for test is to use `it` so that it's closer to language
`it("should process message only once")`